### ISSUE TITLE
[Sort] Sort headers add noise for accessible users when traversing table cells

### ISSUE THREAD

**Issue by fmax32** [angular/components#13012]
#### Bug, feature request, or proposal:
Bug: Sortable table headers add unnecessary noise for accessibility users when traversing table cells.

#### What is the expected behavior?
Grid cells are announced as '<column_name>, <cell_value>, <column_number>'.

#### What is the current behavior?
Grid cells are announced as ''Change Sorting for <column_name>, <cell_value> , <column_number>'.
<img width="1393" alt="screen shot 2018-09-06 at 11 06 47 am" src="https://user-images.githubusercontent.com/6343506/45177565-2bba3800-b1c8-11e8-97b1-9f6247df3fac.png">

#### What are the steps to reproduce?
1. Enable screenreader (for example: VoiceOver on Mac).
2. Review the sorting demo example: https://material.angular.io/components/sort/overview

#### What is the use-case or motivation for changing an existing behavior?
Reducing the cognitive load for screen-reader users.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular: 6.*
Material: 6.*
OS: any
Browser: any

#### Is there anything else we should know?
There is related to a year and 2 days old bug: https://github.com/angular/material2/issues/6874, which is closed as fixed, but it looks like it was partially fixed for traversing table headers only, but there is still an issue with traversing of table cells.

**Comment by jelbourn** [comment]
@crisbeto would you be able to take a look at this with NVDA and see if there's something relatively straightforward we could do? Something having to do with `aria-sort` might be the answer here, but that we'd have to verify on more screen readers.

**Comment by crisbeto** [comment]
@jelbourn I spent some time playing around with it in NVDA and the only way I could get the `aria-sort` to be read out was to remove the `button` inside the `th` and make the `th` itself focusable using `tabindex="0"`. The downsides are that it doesn't read out dynamic changes to sorting (the current approach doesn't do it either) and it seems like introducing a `tabindex` to the `th` breaks navigation using the arrow keys inside the table.

**Comment by jelbourn** [comment]
Yeah, that's not ideal. I'll put this on the list for the next time I talk to the a11y team
